### PR TITLE
Make setup stepdefs works into single language

### PR DESCRIPTION
**Pull request by clicman**

### CODE REVIEW

File: plugins/web-plugin/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/WebSetupStepDefs.java [hunk 1]
```diff
@@ -1,15 +1,20 @@
-package ru.sbtqa.tag.pagefactory.web.stepdefs;
+package ru.sbtqa.tag.pagefactory.stepdefs;
 
-import cucumber.api.java.Before;
 import ru.sbtqa.tag.pagefactory.environment.Environment;
 import ru.sbtqa.tag.pagefactory.web.actions.WebPageActions;
 import ru.sbtqa.tag.pagefactory.web.checks.WebPageChecks;
 import ru.sbtqa.tag.pagefactory.web.drivers.WebDriverService;
 
 public class WebSetupStepDefs {
```
Comment (varivoda) [inline plugins/web-plugin/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/WebSetupStepDefs.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)

File: plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/HtmlSetupStepDefs.java [hunk 2]
```diff
@@ -1,14 +1,19 @@
-package ru.sbtqa.tag.pagefactory.html.stepdefs;
+package ru.sbtqa.tag.pagefactory.stepdefs;
 
-import cucumber.api.java.Before;
 import ru.sbtqa.tag.pagefactory.environment.Environment;
 import ru.sbtqa.tag.pagefactory.html.actions.HtmlPageActions;
 import ru.sbtqa.tag.pagefactory.web.drivers.WebDriverService;
 
 public class HtmlSetupStepDefs {
```
Comment (varivoda) [inline plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/HtmlSetupStepDefs.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)

File: page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/SetupStepDefs.java [hunk 3]
```diff
@@ -15,8 +15,17 @@
 
 public class SetupStepDefs {
```
Comment (varivoda) [inline page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/SetupStepDefs.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)

**Comment by varivoda** [comment]
SonarQube analysis reported 6 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 3 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L118](https://github.com/sbtqa/page-factory-2/blob/967ffaa310bb74d865c61102191a356659101420/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L118): Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L365](https://github.com/sbtqa/page-factory-2/blob/967ffaa310bb74d865c61102191a356659101420/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L365): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![INFO][INFO] [HtmlReflectionUtils.java#L385](https://github.com/sbtqa/page-factory-2/blob/967ffaa310bb74d865c61102191a356659101420/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L385): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/SetupStepDefs.java [hunk 4]
```diff
@@ -15,29 +13,64 @@
...
+        }
+
         TaskHandler.addTask(new ConnectToLogTask());
         TaskHandler.addTask(new KillProcessesTask());
         TaskHandler.addTask(new StartVideoTask());
         TaskHandler.handleTasks();
     }
 
+    public static void setUp(Scenario scenario) {
+        if (isAlreadyPerformed(ispreSetUp)) {
```
Comment (kosteman) [inline page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/SetupStepDefs.java]: isSetUp

File: page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/SetupStepDefs.java [hunk 5]
```diff
@@ -15,29 +13,64 @@
...
         Environment.getDriverService().mountDriver();
         ScenarioContext.setScenario(scenario);
         PageManager.cachePages();
         PageContext.resetContext();
     }
 
-    @After(order = 99999)
-    public void tearDown() {
+    public static void tearDown() {
+        if (isAlreadyPerformed(ispreSetUp)) {
```
Comment (kosteman) [inline page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/SetupStepDefs.java]: isTearDown

File: page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/SetupStepDefs.java [hunk 6]
```diff
@@ -15,29 +13,64 @@
...
-    public void preSetUp(Scenario scenario) {
+    private static final ThreadLocal<Boolean> ispreSetUp = ThreadLocal.withInitial(() -> false);
+    private static final ThreadLocal<Boolean> isSetUp = ThreadLocal.withInitial(() -> false);
+    private static final ThreadLocal<Boolean> isTearDown = ThreadLocal.withInitial(() -> false);
+
+    private SetupStepDefs() {
+        throw new AssertionError("Instantiating utility class.");
+    }
+
+    public static void preSetUp(Scenario scenario) {
```
Comment (varivoda) [inline page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/SetupStepDefs.java]: MAJOR Remove this unused method parameter "scenario". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

**Comment by varivoda** [comment]
SonarQube analysis reported 4 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L118](https://github.com/sbtqa/page-factory-2/blob/e0c5267ace47bc9f421c853e310538e456700236/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L118): Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L365](https://github.com/sbtqa/page-factory-2/blob/e0c5267ace47bc9f421c853e310538e456700236/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L365): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![INFO][INFO] [HtmlReflectionUtils.java#L385](https://github.com/sbtqa/page-factory-2/blob/e0c5267ace47bc9f421c853e310538e456700236/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L385): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by varivoda** [comment]
SonarQube analysis reported 4 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L118](https://github.com/sbtqa/page-factory-2/blob/0d8ff1c0d093b2cba768c75f90711490a5ed5936/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L118): Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L365](https://github.com/sbtqa/page-factory-2/blob/0d8ff1c0d093b2cba768c75f90711490a5ed5936/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L365): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![INFO][INFO] [HtmlReflectionUtils.java#L385](https://github.com/sbtqa/page-factory-2/blob/0d8ff1c0d093b2cba768c75f90711490a5ed5936/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L385): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/SetupSteps.java [hunk 7]
```diff
@@ -13,31 +11,62 @@
...
 
-public class SetupStepDefs {
+public class SetupSteps {
+
+    private static final ThreadLocal<Boolean> isPreSetUp = ThreadLocal.withInitial(() -> false);
+    private static final ThreadLocal<Boolean> isSetUp = ThreadLocal.withInitial(() -> false);
+    private static final ThreadLocal<Boolean> isTearDown = ThreadLocal.withInitial(() -> false);
 
-    @Before(order = 0)
     public void preSetUp(Scenario scenario) {
```
Comment (varivoda) [inline page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/stepdefs/SetupSteps.java]: MAJOR Remove this unused method parameter "scenario". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

**Comment by varivoda** [comment]
SonarQube analysis reported 4 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L118](https://github.com/sbtqa/page-factory-2/blob/7073ee0fba36a8f0d117eb22f63b82679fabd324/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L118): Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L365](https://github.com/sbtqa/page-factory-2/blob/7073ee0fba36a8f0d117eb22f63b82679fabd324/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L365): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![INFO][INFO] [HtmlReflectionUtils.java#L385](https://github.com/sbtqa/page-factory-2/blob/7073ee0fba36a8f0d117eb22f63b82679fabd324/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L385): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by varivoda** [comment]
SonarQube analysis reported 4 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L118](https://github.com/sbtqa/page-factory-2/blob/9c261995e7fec8c2aff79f9464b24ae87e449c6a/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L118): Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L365](https://github.com/sbtqa/page-factory-2/blob/9c261995e7fec8c2aff79f9464b24ae87e449c6a/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L365): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![INFO][INFO] [HtmlReflectionUtils.java#L385](https://github.com/sbtqa/page-factory-2/blob/9c261995e7fec8c2aff79f9464b24ae87e449c6a/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L385): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by varivoda** [comment]
SonarQube analysis reported 4 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L118](https://github.com/sbtqa/page-factory-2/blob/d35314d8c79954adec60585aea42ee4d4dd30ee5/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L118): Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L365](https://github.com/sbtqa/page-factory-2/blob/d35314d8c79954adec60585aea42ee4d4dd30ee5/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L365): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![INFO][INFO] [HtmlReflectionUtils.java#L385](https://github.com/sbtqa/page-factory-2/blob/d35314d8c79954adec60585aea42ee4d4dd30ee5/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L385): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by varivoda** [comment]
SonarQube analysis reported 4 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L118](https://github.com/sbtqa/page-factory-2/blob/388a210de76fe4f87e35f0088e8361f591625e90/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L118): Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [HtmlReflectionUtils.java#L365](https://github.com/sbtqa/page-factory-2/blob/388a210de76fe4f87e35f0088e8361f591625e90/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L365): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![INFO][INFO] [HtmlReflectionUtils.java#L385](https://github.com/sbtqa/page-factory-2/blob/388a210de76fe4f87e35f0088e8361f591625e90/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/html/utils/HtmlReflectionUtils.java#L385): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**